iocore/eventsystem/I_VConnection.h

```diff
@@ -116,13 +116,16 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
-/** Used in VConnection::shutdown(). */
-enum ShutdownHowTo_t {
-  IO_SHUTDOWN_READ = 0,
-  IO_SHUTDOWN_WRITE,
-  IO_SHUTDOWN_READWRITE,
+enum class IOShutdown {
+  READ      = SHUT_RD,
+  WRITE     = SHUT_WR,
+  READWRITE = SHUT_RDWR,
 };
 
+#define IO_SHUTDOWN_READ IOShutdown::READ
```

Comment:
possible tweak if you don't want to say a clean C++ style

`IOShutdown::READ`

is

`constexpr IO_SHUTDOWN_READ  = IOShutdown::READ;`

There is almost no reason in C++ to need a #define

I would just saying 

enum class io_shutdown {
	read= 1,
	write = 2,
	readwrite = 3,
};

and just saying a clean c++ style:

`io_shutdown::read`